hippiehug-package/tests/test_chain.py

```diff
@@ -19,6 +20,28 @@ def test_block_duplicated():
     assert b0.fingers == b1.fingers
     assert b0.aux == b1.aux
 
+
+def test_block_with_hash_duplicated():
+    cc = {
+        'mtr_hash': None,
+        'metadata': {
+            'params': {
+                'dh_pk': 'VVREziRt5sorx9pqi5xpj16t1ibH1XPmNPqXBfCRNZD',
+                'sig_pk': 'VVRF1XiSmteSYFVkBdu9VY588CyPNKymZ6igm9xoEgN',
+                'vrf_pk': 'VVRF2rC2EFimdovJ5yhSx3BLpKsTy7iMTnvsK6NDskB',
+                'rescue_pk': 'VVREycH7jJqvhoEfP7ejAxfz1DAe5cKBUi1zet5cjnt'
+            },
+            'identity_info': "I'm VVREziRt5sorx9pqi5xpj16t1ibH1XPmNPqXBfCRNZD"
+        },
+        'version': 1,
+        'nonce': 'Q3JTBfZr3Q1PZ1JRfx3KNn',
+        'timestamp': 1525089767.931872
+    }
+    b0 = Block(items=[cc], index=0, fingers=[], aux=42)
+    b1 = Block(b0.items, b0.index, b0.fingers, b0.aux)
+    assert b0.hid == b1.hid
```

Comment:
You probably should test for the case `b0.hid != b1.hid` when some of the dict values differ --- to account for the behaviour of `sorted` mentioned above.

Reply:
Thanks. Will start with that test and then make it pass.